fly/commands/internal/executehelpers/inputs.go

```diff
@@ -127,18 +131,30 @@ func TaskInputsContainsName(inputs []atc.TaskInputConfig, name string) bool {
 	return false
 }
 
-func GenerateLocalInputs(fact atc.PlanFactory, inputMappings []flaghelpers.InputPairFlag) (map[string]Input, error) {
+func GenerateLocalInputs(
+	fact atc.PlanFactory,
+	team concourse.Team,
+	inputMappings []flaghelpers.InputPairFlag,
+	includeIgnored bool,
+) (map[string]Input, error) {
+
 	kvMap := map[string]Input{}
 
 	for _, i := range inputMappings {
+		artifact, err := Upload(team, i.Path, includeIgnored)
```

Comment:
I noticed we don't show upload/download progress anymore, now that the steps don't print it. It'd be nice to have that back.

Reply:
erm yeah, i'll look into that

Reply:
I spoke with @xtremerui and he seems to think this got removed a while ago. If we want that back can we create a separate issue?

Reply:
It's there for me:

```
~/w/concourse (fix-dashboard-search) $ fly -t ci e -c ci/tasks/docker-compose-watsjs.yml -i concourse=. -p
executing build 4989469 at https://ci.concourse-ci.org/builds/4989469 
initializing
concourse: 16.54 MiB/s 0s                                                       
running concourse/ci/tasks/scripts/with-docker-compose concourse/ci/tasks/scripts/watsjs
/tmp/build/e55deab7/concourse /tmp/build/e55deab7
Creating network "concourse_default" with the default driver
Pulling db (postgres:)...
```

It's not a progress bar, but it still shows the upload rate and duration.

It shouldn't take long to restore it, and I'd like to avoid user-facing regressions on `master` that prevent us from shipping, so I don't want to merge without it. 😅 

You could probably just hoist the code from the user artifact step into `fly`:

https://github.com/concourse/concourse/blob/b20ee6770be162614a2d7e561c5bff095b81db93/atc/exec/user_artifact_step.go#L54-L62

Reply:
ah cool! im going to pull this into fly execute instead of having it in the step.